apis/exoscale/v1/iam_types.go

```diff
@@ -0,0 +1,142 @@
+package v1
+
+import (
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"reflect"
+)
+
+const (
+	// AccessKeyIDName is the environment variable name for the S3 access key ("username")
+	AccessKeyIDName = "AWS_ACCESS_KEY_ID"
+	// SecretAccessKeyName is the environment variable name for the S3 secret key ("password")
+	SecretAccessKeyName = "AWS_SECRET_ACCESS_KEY"
+)
+
+// SOSSpec is the service type for Object Storage in exoscale
+type SOSSpec struct {
+
+	// +kubebuilder:validation:Required
+	// Buckets is a list of buckets to which IAMKey has access to.
```

Comment:
I would put in a blank line, so that the kubebuilder comment doesn't become part of the field description.
```suggestion
	// +kubebuilder:validation:Required
	
	// Buckets is a list of buckets to which IAMKey has access to.
```